### PR TITLE
MCO: add service monitor to alertmanager

### PR DESCRIPTION
**Pull request by thibaultmg**
Adds platform serviceMonitor for alertManager following [platform monitoring recommendations](https://rhobs-handbook.netlify.app/products/openshiftmonitoring/collecting_metrics.md/):

- Adds serviceMonitor configured with tls
- Adds kube-rbac-proxy sidecar for protecting the metrics endpoint, exposes it on port 9096
- Removes exposed 9093 port from alertmanager service
- Adds new service for port 9096 because existing services have incompatible settings for the metrics endpoint needs (sessionAffinity)
- Adds rbac for fetching configmap in kube-system namespace
- Adds watcher for the configmap in kube-system

As a side note, kube-rbac-proxy automatically reloads TLS and client CA files. So no need to restart the service when they change.

I have tested the manifests and the new serviceMonitor is well being picked up by platform monitoring, and it is able to fetch those new metrics.

Jira task: https://issues.redhat.com/browse/RHOBS-1002

### CODE REVIEW

File: operators/multiclusterobservability/manifests/base/alertmanager/alertmanager-operated.yaml [hunk 1]
```diff
@@ -2,14 +2,10 @@ apiVersion: v1
 kind: Service
 metadata:
   name: alertmanager-operated
-  namespace: open-cluster-management
+  namespace: open-cluster-management-observability
```
Comment (douglascamata) [inline operators/multiclusterobservability/manifests/base/alertmanager/alertmanager-operated.yaml:5]: Is this change necessary? Is it breaking? 🤔

File: operators/multiclusterobservability/manifests/base/alertmanager/alertmanager-statefulset.yaml [hunk 2]
```diff
@@ -138,6 +146,36 @@ spec:
           readOnly: true
         - mountPath: /etc/proxy/secrets
           name: alertmanager-proxy
+      - image: quay.io/brancz/kube-rbac-proxy:v0.15.0
```
Comment (douglascamata) [inline operators/multiclusterobservability/manifests/base/alertmanager/alertmanager-statefulset.yaml:149]: We can't rely on this image because of compliance. Need to use `quay.io/stolostron/kube-rbac-proxy`

**Review by douglascamata — CHANGES_REQUESTED** [review]

File: operators/multiclusterobservability/manifests/base/alertmanager/alertmanager-operated.yaml (hunk 1 above)
Comment (thibaultmg) [reply operators/multiclusterobservability/manifests/base/alertmanager/alertmanager-operated.yaml:5]: All those namespaces are reset by the controller. So personally, I would remove all of them from the manifests. But I am not sure it has not impact 🤷 
Resetting this one as it wasn't changed on purpose

**Comment by douglascamata** [comment]
@thibaultmg something's badly wrong with Prow, it seems.

File: operators/multiclusterobservability/pkg/rendering/renderer_alertmanager.go [hunk 3]
```diff
@@ -145,6 +158,55 @@ func (r *MCORenderer) renderAlertManagerSecret(res *resource.Resource,
...
+		if !exists {
+			return nil, fmt.Errorf("client-ca-file not found in source ConfigMap")
+		}
+
+		if len(caData) == 0 {
+			return nil, fmt.Errorf("client-ca-file is empty in source ConfigMap")
+		}
+
+		// Update the ConfigMap with the CA certificate data
+		cm.Data["client-ca-file"] = caData
```
Comment (saswatamcode) [inline operators/multiclusterobservability/pkg/rendering/renderer_alertmanager.go:197]: Wouldn't this result in a panic, as we are assigning to nil map?

File: operators/multiclusterobservability/manifests/base/alertmanager/alertmanager-operated.yaml (hunk 1 above)
Comment (thibaultmg) [reply operators/multiclusterobservability/manifests/base/alertmanager/alertmanager-operated.yaml]: I removed all unnecessary namespaces. Unit tests are checking them now

File: operators/multiclusterobservability/manifests/base/alertmanager/alertmanager-accessor-serviceaccount.yaml [hunk 4]
```diff
@@ -2,6 +2,5 @@ apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: observability-alertmanager-accessor
-  namespace: open-cluster-management
```
Comment (douglascamata) [inline operators/multiclusterobservability/manifests/base/alertmanager/alertmanager-accessor-serviceaccount.yaml:5]: Is this expected? In which namespace will this land?

**Review by douglascamata — APPROVED** [review]
LGTM :shipit:

File: operators/multiclusterobservability/manifests/base/alertmanager/alertmanager-accessor-serviceaccount.yaml (hunk 4 above)
Comment (thibaultmg) [reply operators/multiclusterobservability/manifests/base/alertmanager/alertmanager-accessor-serviceaccount.yaml:5]: Service account resources are handled by `"ServiceAccount":        r.renderer.RenderNamespace,`
So it will land as before in the `namespace := mcoconfig.GetDefaultNamespace()` which is `open-cluster-management-observability`
I removed all of them so that there is no ambiguity. And they are all checked in unit tests.

**Review by douglascamata — APPROVED** [review]
